Issue  4654  Updates to tickets/ticket48234_test.py

kimettog opened the pull request:
[IDMDS-1068] Update ticket48234_test.py rename and move to --> suites/acl/aci_excl_filter_test.py
            - Update to use DSLdapObject class

kimettog committed:
IDMDS-1068 Update failing ticket48234_test.py test

mreynolds389 committed:
Issue 4149 - UI - port TreeView and opther components to PF4

Description:  This ports all th TreeViews to PF4, and also does some proof
              of concept changes for PF3 to PF4 migration.  There is much
              more needed, but this does not break anything

relates: https://github.com/389ds/389-ds-base/issues/4149

Reviewed by: spichugi(Thanks!)

mreynolds389 committed:
Issue 4609 - CVE - info disclosure when authenticating

Description:  If you bind as a user that does not exist.  Error 49 is returned
              instead of error 32.  As error 32 discloses that the entry does
              not exist.  When you bind as an entry that does not have userpassword
              set then error 48 (inappropriate auth) is returned, but this
              discloses that the entry does indeed exist.  Instead we should
              always return error 49, even if the password is not set in the
              entry.  This way we do not disclose to an attacker if the Bind
              DN exists or not.

Relates: https://github.com/389ds/389-ds-base/issues/4609

Reviewed by: tbordaz(Thanks!)

mreynolds389 committed:
Issue 4513 - CI - make acl ip address tests more robust

Description:  The tests aumme the system is using IPv6 loopback address, but it
              should still check for IPv4 loopback.

Relates: https://github.com/389ds/389-ds-base/issues/4513

Reviewed by: ?

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

vashirov committed:
Issue 4577 - Add GitHub actions

Description:

* Enable IPv6 support for docker daemon
* Set server.example.com as FQDN for container

Relates: https://github.com/389ds/389-ds-base/issues/4577

Reviewed by: @droideck (Thanks!)

progier389 committed:
issue 4612 - Fix pytest fourwaymmr_test for non root user (#4613)

kimettog committed:
Issue 4654 - Updates to tickets/ticket48234_test.py

Bug Description:

Update to tickets/ticket48234_test.py which are currently failing and using
soon to be obsolete classes

Fix Description:

Updated tickets/ticket48234_test.py and ported to the suites directory
Updated to utilise the DSLDAPObject class methods

relates: <The Issue URL>

Author: Gilbert Kimetto

Reviewed by: ???
IDMDS-1068 Update failing ticket48234_test.py test

[INTEROP-4009] CodeReady Studio on OpenShift - Run locally

[INTEROP-4009] CodeReady Studio on OpenShift - Run locally

[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
IDMDS-1068 Update failing ticket48234_test.py test

kimettog committed:
[INTEROP-4009] CodeReady Studio on OpenShift - Run locally

Firstyear committed:
Issue 4591 - RFE - improve openldap_to_ds help and features (#4607)

Bug Description: Improve the --help page, and finish wiring in some
features.

Fix Description: Wire in exclusion of attributes/schema for migration.

fixes: https://github.com/389ds/389-ds-base/issues/4591

Author: William Brown <william@blackhats.net.au>

Review by: @mreynolds389, @droideck

Jackbennett committed:
Update dscontainer (#4564)

Issue 4564 - RFE - Add suffix to dscontainer rc file

Bug Description: The suffix was not added before, adding a hurdle to
automatic admin of the container instance

Fix Description: If the suffix is set, add it to the created rc file. 

fixes: https://github.com/389ds/389-ds-base/pull/4564

Author: @Jackbennett

Review by: @Firstyear

progier389 committed:
Issue 4469 - Backend redesign phase 3a - bdb dependency removal from back-ldbm

A massive change (https://directory.fedoraproject.org/docs/389ds/design/backend-redesign-phase3.html) that implements and use the dbimpl API in the backend.

droideck committed:
Issue 4593 - RFE - Print help when nsSSLPersonalitySSL is not found (#4614)

Description: RHDS instance will fail to start if the TLS server
certificate nickname doesn't match the value of the configuration
parameter "nsSSLPersonalitySSL".

The mismatch typically happens when customers copy the NSS DB from
a previous instance or export the certificate's data but forget to set
the "nsSSLPersonalitySSL" value accordingly.

Log an additional message which should help a user to set up
nsSSLPersonalitySSL correctly.

Fixes: #4593

Reviewed by: @Firstyear (Thanks!)

mreynolds389 committed:
Issue 4324 - Some architectures the cache line size file does not exist

Bug Description:  When optimizing our mutexes we check for a system called
                  coherency_line_size that contains the size value, but if
                  the file did not exist the server would crash in PR_Read
                  (NULL pointer for fd).

Fix Description:  Check PR_Open() was successfully before calling PR_Read().

Relates: https://github.com/389ds/389-ds-base/issues/4324

Reviewed by: tbordaz(Thanks!)

mreynolds389 committed:
Issue 4615 - log message when psearch first exceeds max threads per conn

Desciption:  When a connection hits max threads per conn for the first time
             log a message in the error.  This will help customers diagnosis
             misbehaving clients.

Fixes: https://github.com/389ds/389-ds-base/issues/4615

Reviewed by: progier389(Thanks!)

progier389 committed:
Issue 4469 - Backend redesing phase 3a - implement dbimpl API and use it in back-ldbm (#4618)

see design document https://directory.fedoraproject.org/docs/389ds/design/backend-redesign-phase3.html

mreynolds389 committed:
Issue 4619 - remove pytest requirement from lib389

Description:  Remove the requirement for pytest from lib389, it causes
              unneeded package requirements on Fedora/RHEL.

Fixes: https://github.com/389ds/389-ds-base/issues/4619

Reviewed by: mreynolds(one line commit rule)

mreynolds389 committed:
Bump version to 2.0.3

kimettog committed:
[INTEROP-4009] CodeReady Studio on OpenShift - Run locally

kimettog committed:
[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

bsimonova committed:
Issue 2820 - Fix CI test suite issues

Description:
tickets/ticket48961_test.py was failing in CI nightly runs.
Fixed the failure by changing the code to use DSLdapObject
and moved the code into the config test suite.

Relates: https://github.com/389ds/389-ds-base/issues/2820

Reviewed by: droideck (Thanks!)

mreynolds389 committed:
Issue 4169 - UI - port charts to PF4

Description:  Ported the charts under the monitor tab to use PF4 sparkline charts
              and provide realtime stats on the the caches.

Relates: https://github.com/389ds/389-ds-base/issues/4169

Reviewed by: spichugi(Thanks!)

mreynolds389 committed:
Issue 4169 - UI - Migrate Accordians to PF4 ExpandableSection

Description:  Replace all the CustomCollapse components with PF4
              ExpandableSection component.

relates: https://github.com/389ds/389-ds-base/issues/4169

Reviewed by: spichugi(Thanks!)

[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

mreynolds389 committed:
Issue 4169 - UI - Migrate alerts to PF4

Description:  Migrate the toast notifications to PF4 Alerts.

              Also fixed a refresh problem on the Tuning page.

relates: https://github.com/389ds/389-ds-base/issues/4169

Reviewed by: spichugi(Thanks!)

jchapma committed:
Issue 4595 - Paged search lookthroughlimit bug (#4602)

Bug Description: During a paged search with lookthroughlimit enabled,
lookthroughcount is used to keep track of how many entries are
examined. A paged search reads ahead one entry to catch the end of the
search so it doesn't show the prompt when there are no more entries.
lookthroughcount doesn't take read ahead into account when tracking
how many entries have been examined.

Fix Description: Keep lookthroughcount in sync with read ahead by
by decrementing it during read ahead roll back.

Fixes: https://github.com/389ds/389-ds-base/issues/4595

Relates: https://github.com/389ds/389-ds-base/issues/4513

Reviewed by: droideck, mreynolds389, Firstyear, progier389 (Many thanks)

mreynolds389 committed:
Issue 4646 - CLI/UI - revise DNA plugin management

Bug Description:

There was a false assumption that you have to create the shared DNA
server configuration entry, but in fact the server creates and manages
this entry.  The only thing you should edit in this entry are the
remote Bind Method and Connection Protocol.

Fix Description:

Remove the options to create the shared config entry, and edit the
core/reserved attributes.

Also fixed some issues where we were not showing CLI plugin output in
proper JSON.  This required some changes to the UI as well.

Relates: https://github.com/389ds/389-ds-base/issues/4646

Reviewed by: spichugi(Thanks!)

[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

tbordaz committed:
Issue 4649 - crash in sync_repl when a MODRDN create a cenotaph (#4652)

Bug description:
	When an operation is flagged OP_FLAG_NOOP, it skips BETXN plugins but calls POST plugins.
	For sync_repl, betxn (sync_update_persist_betxn_pre_op) creates an operation extension to be
	consumed by the post (sync_update_persist_op). In case of OP_FLAG_NOOP, there is no
	operation extension.

Fix description:
	Test that the operation is OP_FLAG_NOOP if the operation extension is missing

relates: https://github.com/389ds/389-ds-base/issues/4649

Reviewed by: William Brown (thanks)

Platforms tested: F31

tbordaz committed:
Issue 4644 - Large updates can reset the CLcache to the beginning of the changelog (#4647)

Bug description:
	The replication agreements are using bulk load to load updates.
	For bulk load it uses a cursor with DB_MULTIPLE_KEY and DB_NEXT.
	Before using the cursor, it must be initialized with DB_SET.

	If during the cursor/DB_SET the CSN refers to an update that is larger than
	the size of the provided buffer, then the cursor remains not initialized and
	c_get returns DB_BUFFER_SMALL.

	The consequence is that the next c_get(DB_MULTIPLE_KEY and DB_NEXT) will return the
	first record in the changelog DB. This break CLcache.

Fix description:
	The fix is to harden cursor initialization so that if DB_SET fails
	because of DB_BUFFER_SMALL. It reallocates buf_data and retries a DB_SET.
	If DB_SET can not be initialized it logs a warning.

	The patch also changes the behaviour of the fix #4492.
	#4492 detected a massive (1day) jump prior the starting csn and ended the
	replication session. If the jump was systematic, for example
	if the CLcache got broken because of a too large updates, then
	replication was systematically stopped.
	This patch suppress the systematically stop, letting RA doing a big jump.
	From #4492 only remains the warning.

relates: https://github.com/389ds/389-ds-base/issues/4644

Reviewed by: Pierre Rogier (Thanks !!!!)

Platforms tested: F31

kimettog committed:
[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

kimettog committed:
[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog commented on dirsrvtests/tests/suites/acl/aci_excl_filter_test.py:
```diff
@@ -0,0 +1,153 @@
+from lib389.idm.user import UserAccounts
+from lib389._constants import DEFAULT_SUFFIX, DN_DM, PW_DM
+
+pytestmark = pytest.mark.tier2
```
Should this be updated to tier 1 test

droideck committed:
Issue 4643 - Add a tool that generates Rust dependencies for a specfile (#4645)

Description: The Fedora builds of 389-DS uses the vendored crates
to build the official packages for Rawhide. Vendoring and bundling
dependencies is in violation of Fedora policies. As an upstream project
we are free to ship vendored code. But as a downstream Fedora project
we must not use the vendored code.

Add a tool that will help to generate 'Provides: bundled(crate(foo)) = version'
for Cargo.lock file content.
Replace License field which should contain all of the package licenses
we bundle in the specfile.

Fixes: https://github.com/389ds/389-ds-base/issues/4643

Reviewed by: @Firstyear, @decathorpe, @mreynolds389 (Thanks!)

kimettog committed:
[IDMDS-1068] Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

kimettog commented on dirsrvtests/tests/suites/acl/aci_excl_filter_test.py:
```diff
@@ -0,0 +1,146 @@
+    cn_filter = '(cn=%s)' % username
+    try:
+        # entries = topo.standalone.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, cn_filter, [deniedattr, 'dn'])
+        entries = user.get((DEFAULT_SUFFIX, cn_filter, [deniedattr, 'dn']))
```
I am getting a failure here with entries...I am expecting to get a list of length 2

kimettog replied:
Resolved

bsimonova commented on dirsrvtests/tests/suites/acl/aci_excl_filter_test.py:
```diff
@@ -0,0 +1,146 @@
+                 '(version 3.0;acl "admin-tel-matching-rule-outest";deny (all)' +
+                 '(userdn = "ldap:///{}??sub?(&(cn={})(ou:dn:={}))");)'.format(DEFAULT_SUFFIX, username, ouname))
+    try:
+         topo.standalone.modify_s(DEFAULT_SUFFIX, [(ldap.MOD_ADD, 'aci', ensure_bytes(aci_text))])
```
I think you can use Domain to add aci instead of modify_s:

`from lib389.idm.domain import Domain`
`Domain(topo.standalone, DEFAULT_SUFFIX).add("aci", aci_text)`

There is usage in suites/acl/acivattr_test.py

But I am working with aci for the first time so I would like someone else to confirm.
@vashirov Can I ask you to also review, please? Thanks!

progier389 committed:
issue 4552 - Backup Redesign phase 3b - use dbimpl in replicatin plugin (#4622)

* issue 4552 - Backup Redesign phase 3b - use dbimpl in replicatin plugin

Merge of a fix in cl5_clcache.c (changelog cache restarts from begining if large update)
Rebase with master

* Issue 4469 - Backend redesing phase 3a - implement dbimpl API and use it in back-ldbm - fix test_maxbersize_repl pytest failure

* issue 4552 - Backup Redesign phase 3b - use dbimpl in replicatin plugin - fix indent issue

* issue 4552 - Backup Redesign phase 3b - use dbimpl in replicatin plugin - fix merge issue

manual Merge of fix about changelog cache iteration restarting from beginning in case of large update + automatic rebase to master

* Issue 4552 - Backend redesign phase 3b - fix indent issue + random crash and memory leak in tombstone handling

mreynolds389 committed:
Merge pull request #4664 from mreynolds389/issue4663

Issue 4663 - CLI - unable to add objectclass/attribute without x-origin

Issue 4654 Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

kimettog committed:
Issue 4654 Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

kimettog committed:
Merge branch 'master' of github.com:kimettog/389-ds-base

kimettog committed:
Merge branch 'master' of github.com:389ds/389-ds-base

kimettog committed:
Issue 4654 - Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

Bug Description:

- Update ticket48234_test.py to verify tests on RHEL 7/8 and Fedora
- Update deprecated "*_s" methods to leverage the DSLDAPObject class
- Move test from the current location in ../tickets to appropriate ../suites/aci/* directory

Fix Description:
- Issue 4654 Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

relates:

Author: Gilbert Kimetto

Reviewed by: ???

kimettog committed:
Issue 4654 Update ticket48234_test.py and move to suites/acl/aci_excl_filter_test.py

droideck commented on dirsrvtests/tests/suites/acl/aci_excl_filter_test.py:
```diff
@@ -0,0 +1,150 @@
+
+
+def test_aci_with_exclude_filter(topo, add_anon_aci_access):
+    """
```
The docstring should start here - something like this:

```
"""Test description

:id: IDDDD
:setup: ....
...
"""
```

kimettog replied:
thank you fixed the docstring to  (see below)

def test_aci_with_exclude_filter(topo, add_anon_aci_access):
    """Test an ACI(Access control instruction) which contains an extensible filter.
    :id: 238da674-81d9-11eb-a965-98fa9ba19b65

droideck commented on dirsrvtests/tests/suites/acl/aci_excl_filter_test.py:
```diff
@@ -0,0 +1,150 @@
+        1. Operation should be successful
+        2. Operation should be successful
+        3. Operation should be successful
+        4. PASS - users found do not have the --> deniedattr = 'telephonenumber' marker
```
You have only 4 'expectedresults' but you have 7 'steps'.

I think our test documentation system requires a full alignment (7 steps - 7 expected results)

kimettog replied:
Thank you -fixed the missing results.

droideck reviewed (approved):
Besides these small docstring issues - Looks Good To Me!

kimettog committed:
Merge branch 'master' of github.com:kimettog/389-ds-base

droideck reviewed (approved):
Looks good!
